arm/Microsoft.SignalRService/webPubSub/deploy.bicep

```diff
@@ -0,0 +1,161 @@
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource webPubSub 'Microsoft.SignalRService/webPubSub@2021-10-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    capacity: capacity
+    name: sku
+    tier: sku == 'Standard_S1' ? 'Standard' : 'Free'
+  }
+  identity: identity
+  properties: {
+    disableAadAuth: disableAadAuth
+    disableLocalAuth: disableLocalAuth
+    networkACLs: !empty(webPubSubNetworkAcls) ? {
+    } : null
+    publicNetworkAccess: publicNetworkAccess
+    resourceLogConfiguration: {
+      categories: resourceLogConfiguration
+    }
+    tls: {
+      clientCertEnabled: clientCertEnabled
+    }
+  }
+}
+
+module webPubSub_privateEndpoints '../../Microsoft.Network/privateEndpoints/deploy.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
+  name: '${uniqueString(deployment().name, location)}-appConfiguration-PrivateEndpoint-${index}'
+  params: {
+    groupIds: [ privateEndpoint.service ]
+    name: contains(privateEndpoint, 'name') ? privateEndpoint.name : 'pe-${last(split(webPubSub.id, '/'))}-${privateEndpoint.service}-${index}'
+    serviceResourceId: webPubSub.id
+    subnetResourceId: privateEndpoint.subnetResourceId
+    location: reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
+    lock: contains(privateEndpoint, 'lock') ? privateEndpoint.lock : lock
+    privateDnsZoneGroups: contains(privateEndpoint, 'privateDnsZoneGroups') ? privateEndpoint.privateDnsZoneGroups : []
+    roleAssignments: contains(privateEndpoint, 'roleAssignments') ? privateEndpoint.roleAssignments : []
+    tags: contains(privateEndpoint, 'tags') ? privateEndpoint.tags : {}
+    manualPrivateLinkServiceConnections: contains(privateEndpoint, 'manualPrivateLinkServiceConnections') ? privateEndpoint.manualPrivateLinkServiceConnections : []
+    customDnsConfigs: contains(privateEndpoint, 'customDnsConfigs') ? privateEndpoint.customDnsConfigs : []
+  }
+}]
+
+resource webPubSub_lock 'Microsoft.Authorization/locks@2017-04-01' = if (!empty(lock)) {
+  name: '${webPubSub.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: webPubSub
+}
+
+module webPubSub_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${uniqueString(deployment().name, location)}-AppConfig-Rbac-${index}'
+  params: {
+    description: contains(roleAssignment, 'description') ? roleAssignment.description : ''
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceId: webPubSub.id
+  }
+}]
```

Comment:
The [default outputs] (https://github.com/Azure/ResourceModules/wiki/The%20library%20-%20Module%20design#outputs) are missing. Please add